File: algorithms/solutions/week23/huytd/71.simplify-path.rs
/*
 * [71] Simplify Path
 *
 * https://leetcode.com/problems/simplify-path/description/
 *
 * Status:
 *  ✔ Accepted
 *  ✔ 252/252 cases passed (60 ms)
 */

fn simplify_path(path: &str) -> String {
    let mut stack = vec![""];
    let tokens = path.split("/");
    for token in tokens {
        if token.ne("") && token.ne(".") {
            if token.eq("..") {
                stack.pop();
            } else {
                stack.push(token);
            }
        }
    }
    stack.join("/")
}

fn main() {
    assert!(simplify_path("/home/a/./c/../d///e") == "/home/a/d/e");
}

File: algorithms/solutions/week23/huytd/71.simplify-path.js
/*
 * [71] Simplify Path
 *
 * https://leetcode.com/problems/simplify-path/description/
 *
 * Status:
 *  ✔ Accepted
 *  ✔ 252/252 cases passed (60 ms)
 *  ✔ Your runtime beats 97.1 % of javascript submissions
 */

var simplifyPath = function(path) {
    let stack = [];
    path.split("/").forEach(token => {
        if (token !== "" && token !== ".") {
            if (token === "..") {
                stack.pop();
            } else {
                stack.push(token);
            }
        }
    });
    return "/" + stack.join("/");
};

//console.log(simplifyPath("/home/a/./c/../d///e"));
